wallet-new/src/Cardano/Wallet/WalletLayer/Legacy.hs

```
@@ -1,43 +1,233 @@
+-- the DB.
+pwlDeleteWallet
+    :: forall ctx m. (MonadLegacyWallet ctx m)
+    => WalletId
+    -> m Bool
+pwlDeleteWallet wId = do
+    cWId        <- migrate wId
+    -- TODO(ks): It would be better to catch specific @Exception@.
+    -- Maybe @try@?
+    catchAll (const True <$> V0.deleteWallet cWId) (const . pure $ False)
+
+------------------------------------------------------------
+-- Account
+------------------------------------------------------------
+
+pwlCreateAccount
+    :: forall ctx m. (MonadLegacyWallet ctx m)
+    => WalletId
+    -> NewAccount
+    -> m Account
+pwlCreateAccount wId newAcc@NewAccount{..} = do
+
+    let spendingPassword = fromMaybe mempty . fmap coerce $ naccSpendingPassword
+
+    accInit     <- migrate (wId, newAcc)
+    cAccount    <- V0.newAccount RandomSeed spendingPassword accInit
+
+    migrate cAccount
+
+pwlGetAccounts
+    :: forall ctx m. (MonadLegacyWallet ctx m)
+    => WalletId
+    -> m [Account]
+pwlGetAccounts wId = do
+    cWId        <- migrate wId
+    cAccounts   <- V0.getAccounts $ Just cWId
+    migrate cAccounts
+
+pwlGetAccount
+    :: forall ctx m. (MonadLegacyWallet ctx m)
+    => WalletId
+    -> AccountIndex
+    -> m (Maybe Account)
+pwlGetAccount wId aId = do
+    accId       <- migrate (wId, aId)
+    account     <- V0.getAccount accId
+    Just <$> migrate account
+
+pwlUpdateAccount
+    :: forall ctx m. (MonadLegacyWallet ctx m)
+    => WalletId
+    -> AccountIndex
+    -> AccountUpdate
+    -> m Account
+pwlUpdateAccount wId accIdx accUpdate = do
+    newAccId    <- migrate (wId, accIdx)
+    accMeta     <- migrate accUpdate
+    cAccount    <- V0.updateAccount newAccId accMeta
+    migrate cAccount
+
+pwlDeleteAccount
+    :: forall ctx m. (MonadLegacyWallet ctx m)
+    => WalletId
+    -> AccountIndex
+    -> m Bool
+pwlDeleteAccount wId accIdx = do
+    accId <- migrate (wId, accIdx)
+    catchAll (const True <$> V0.deleteAccount accId) (const . pure $ False)
```

Same remark as above. It feels weird to swallow exceptions here -- especially when we don't do it for other methods.